### DISCUSSION TITLE
[feature request] clear settings search box after exiting settings

### DISCUSSION THREAD

**Discussion by rszyma** [ppy/osu#24470] [Ideas] [answered]
It's annoying that literally *every time* I press Ctrl+O I have clear the previous search! If you don't want to make this behavior default at least make it optional. Or at the *very least* clear it after user have not entered settings for a minute or so. Please!

**Comment by Theighlin** [comment]
You can just press escape twice (first clears search box, second closes overlay)

**Reply by rszyma** [reply]
Thanks for suggestion. However this doesn't really solve the problem, because when I press Ctrl+O I expect to always be able to immediately search for new thing, regardless of how I exited the settings (with double Esc, or by clicking outside settings)

**Comment by rszyma** [comment]
I came up with an alternative feature suggestion that would solve this problem:

When pressing Ctrl+O select all the previous text in the search box. It would work similar to how Ctrl+L works in browser (jumps to navigation bar and selects all).

**Reply by peppy** [reply]
Sounds good, but will depend on https://github.com/ppy/osu/discussions/24481.

**Reply by peppy** [reply, marked as answer]
https://github.com/ppy/osu/issues/24606
